src/main/java/net/minecraftforge/network/packets/OpenContainer.java

```diff
@@ -45,6 +45,7 @@ public static void encode(OpenContainer msg, FriendlyByteBuf buf) {
         buf.writeVarInt(msg.windowId);
         buf.writeComponent(msg.name);
         buf.writeByteArray(msg.additionalData.readByteArray());
+        msg.additionalData.resetReaderIndex();
```

Comment:
You're missing the markReaderIndex

Reply:
When there's no reader index marked, it just resets the reader index back to 0, which should be fine in this case. I guess it might lead to obscure issues when something messes with the buffer beforehand, so I will add it just to be safe.

Reply:
Always good to be safe. The other option is to use `buf.writeBytes(data, data.readerIndex(), data.readableBytes())` which doesnt modify the indexes at all. Or you could of done `write(data.slice())` so it had its own indexes.
But this was is simple as well.

Reply:
Ah good to know, I was indeed wondering why some places used `data.slice()` instead of just passing in the buffer itself, now I know. Thanks for the tips!